lib/order.rb

```diff
@@ -0,0 +1,53 @@
+require_relative 'menu'
+require_relative 'sms'
+
+class Order
+  attr_reader :basket
+
+  def initialize 
+    @basket = []
+    #eventually would be equal to a premade menu object (i.e Italian_restaurant) or would be able to select a premade menu
+    @menu = Menu.new   
+  end
+
+  def add_dish_to_basket(dish)
+    count = 0
+    @menu.menu.each { |food| @basket << food && count +=1 if food.name == dish }
+    raise 'There is no such dish on the menu' if count == 0
+  end
+
+  def remove_dish_from_basket(dish) 
+    matching_dishes = @basket.select { |food| food.name == dish}
+    if matching_dishes.count >= 1 
+      @basket.delete_at(@basket.index(matching_dishes[0])) 
+    else
+      raise "This dish is not currently in your basket"
```

Comment:
Minor one - for readability, prefer "short-circuit" guards like this, so you don't have to write an indented if/else statement:
```ruby
raise "This dish is not currently in your basket" if matching_dishes.empty?

@basket.delete_at(@basket.index(matching_dishes[0]))
```